### ISSUE TITLE
Missing simple type serializer for Visual Studio 2015 compilation in serialization.h

### ISSUE THREAD

**Issue by fduguet** [mariusmuja/flann#298]
Hi,

While building flann 1.8.4 using CMake and Visual Studio 2015 (Win64) community edition I got an error in the build. The message was reading that some type did not have an implementation for serialize.

I added size_t to the list of basic type serializer @ line 92 of serialization.h:

`BASIC_TYPE_SERIALIZER(size_t);`

And the whole build went through.

Is it a configuration issue, or a corner case ?

**Comment by PeterSchuebel** [comment]
Hi.
I just noticed this as well in MSVC 2013 WIN64 (built project files with CMake 3.7.1).
Thanks for the hint at fixing it. Instead of using _size_t_ I added
`BASIC_TYPE_SERIALIZER(unsigned long long);`
because that seemed more appropriate.
I think that the issue is that on WIN64 platforms _unsigned int_ and _unsigned long_ are both 32-bit types (unlike on some other 64-bit platforms). So a more explicit _unsigned long long_ is needed.
The Win32 configurations built fine for me before this change, probably because the code that actually used this is using _size_t_. I didn't look at that further, but in my mind serialising size_t can be a bug as well, e.g. if you expect to read the data from 32-bit as well as 64-bit binaries.

It would be nice if one of the project maintainers could reply to this issue here.
Thanks,
Peter
